### PR TITLE
Add CircleCI build and publish

### PR DESCRIPTION
**Pull request by rtfpessoa**
:breaking:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    steps:
+      - checkout
+      - run:
+          name: Get version
+          command: |
+            export VERSION_NUMBER=$(make get-next-version-number)
+            echo $VERSION_NUMBER >> ~/workdir/.version
+      - persist_to_workspace:
+          root: ~/
+          paths:
```
Comment (andreaTP) [inline .circleci/config.yml]:
```
      - persist_to_workspace:
          root: ~/
          paths:
            - workdir/*
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  tag_version:
+    <<: *default_machine_job
+    steps:
+      - attach_workspace:
+          at: ~/
+      - *set_version_env_variable
+      - add_ssh_keys:
+          fingerprints:
+            - "df:83:d7:c7:d5:79:06:c2:3b:d1:fd:e2:a3:d1:12:c5"
```
Comment (andreaTP) [inline .circleci/config.yml]: check with @h314to  that this is still valid

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+.DEFAULT-GOAL := help
+
+# Variables
+PROJECT_NAME?=$$(git remote -v | head -n1 | awk '{print $$2}' | sed 's/.*\///' | sed 's/\.git//')
+CWD=$(shell pwd)
+VERSION_NUMBER?=$$(docker run -v $(CWD):/repo codacy/git-version:latest)## lazy. run with VERSION_NUMBER=x to prevent slow runs
+
+.PHONY: help get-next-version-number build push-docker-image push-latest-docker-image
+help:
+	@echo "---------------------------------------------------------------------------------------------------------"
```
Comment (andreaTP) [inline Makefile]: remove fixed length strings, they play really bad with terminal resize

File: .circleci/config.yml (hunk 2 above)
Comment (rtfpessoa) [reply .circleci/config.yml]: it is, I checked

File: Makefile (hunk 3 above)
Comment (rtfpessoa) [reply Makefile]: done

File: .circleci/config.yml (hunk 1 above)
Comment (rtfpessoa) [reply .circleci/config.yml]: done

**Pull request merged**